pyveda/frameworks/batch_generator.py

```diff
@@ -43,148 +74,131 @@ def mltype(self):
+    def shuffle_ids(self):
+        ''' shuffle the ID lookup table '''
+        np.random.shuffle(self.id_lut)
+
+    def apply_augmentations(self, x, y):
+        """ Applies all built-in transforms, returns augmented x/y
+            Args:
+              x: image array
+              y: label
         """
         augmentation_list = []
-        if self.flip_h:
+
+        if self.flip_h and choice([True, False]):
             augmentation_list.append(np.fliplr)
-        if self.flip_v:
+        if self.flip_v and choice([True, False]):
             augmentation_list.append(np.flipud)
 
-        if len(augmentation_list) == 0:
-            return augmentation_list
-        else:
-            # randomly select augmentations to perform
-            randomly_selected_functions_lst = sample(augmentation_list, k=randint(0, len(augmentation_list)))
-        return randomly_selected_functions_lst
+        for t_fn in augmentation_list:
+            x = t_fn(x)
 
-    def apply_transforms(self, x, y):
-        """
-            Applies a transform method, returns x/y augmentations
-            Args:
-              x: image array
-              y: label
-        """
-        # Figure out which transform to apply randomly
-        transforms = self._applied_augs(self.flip_h, self.flip_v)
-
-        # User selects no augmentation functions
-        if len(transforms) == 0:
-                if self.mltype == 'object_detection':
-                    return x, y
-                return x, y
-
-        # Make Augmentations
-        if self.mltype == 'classification':
-            for t_fn in transforms:
-                x = t_fn(x)
-            return x, y
-
-        if self.mltype == 'object_detection':
-            for t_fn in transforms:
-                x = t_fn(x)
+            if self.mltype == 'segmentation':
+                y = t_fn(y)
+
+            elif self.mltype == 'object_detection':
                 if t_fn is np.fliplr:
                     y = flip_labels_horizontal(self.shape, y)
-                if t_fn is np.flipud:
+                elif t_fn is np.flipud:
                     y = flip_labels_vertical(self.shape, y)
-            return x, y
-
-        if self.mltype == 'segmentation':
-            for t_fn in transforms:
-                x = t_fn(x)
-                y = t_fn(y)
-            return x, y
+        
+        return x, y
 
-    def build_batch(self, index):
-        raise NotImplemented
+    def build_batch(self, step):
+        raise NotImplementedError
 
-    def __getitem__(self, index):
-        return self.build_batch(index)
+    def __getitem__(self, batch):
+        raise NotImplementedError("Getting a specific batch is not supported since batches are generally randomized")
 
     def __next__(self):
-        try:
-            item = self[self.index]  # index???
-        except IndexError:
-            raise StopIteration
-        self.index += 1
-        return item
-
-    def on_epoch_end(self):
```

Comment:
Based on my research `on_epoch_end` is for Keras image *iterators*, not batch *generators*. So I took it out. But if the concepts end up bleeding into each other then we might need to add a hook for this back in.

Reply:
ok cool, good to know